c7n/filters/metrics.py

```diff
@@ -125,11 +126,19 @@ class MetricsFilter(Filter):
         'workspaces': 'AWS/WorkSpaces',
     }
 
+    standard_stats = {'Average', 'Sum', 'Maximum', 'Minimum', 'SampleCount'}
+    extended_stats_re = re.compile(r'^p\d{1,3}\.{0,1}\d{0,1}$')
```

Comment:
I think the decimal bits should be optional here to cover cases like `p95`. I realize we're making a tradeoff here to catch common whiffs rather than being exhaustive, but wonder if doing a float conversion and checking for `0 <= x <= 100` is more or less annoying than the regex?

Reply:
I actually implemented the non regex check, but it ended up being about a dozen lines and it felt debatable on the value, the regex isn't perfect but it captures enough validation for most purposes (everything except the over 100 range check, but that seems like it would stick out pretty clearly in a policy review). the decimal spot is optional in the regex, re p95 works.

Reply:
Ha, didn't see question marks and looked right past the 0,1 :facepalm: . Cool, that works :+1: